app/models/tenant.rb

```diff
@@ -280,6 +281,10 @@ def create_tenant_group
     self
   end
 
+  def ensure_can_be_destroyed
+    raise "A tenant with groups associated cannot be deleted." unless miq_groups.non_tenant_groups.empty?
```

Comment:
If I recall correctly, `.empty?` on a relation results in a `SELECT COUNT(*) ...` query. On the other hand, `.exists?` yields `SELECT 1 AS one ... LIMIT 1`, which is probably faster. Is it possible to change this `unless ... empty?` to `if ... exists?`